pkgs/servers/draupnir/default.nix

```diff
@@ -0,0 +1,97 @@
+{ lib
+, nixosTests
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+, nodejs
+, pkgs
+, mkYarnPackage
+, fetchYarnDeps
+}:
+
+# docs: https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/javascript.section.md#yarn2nix-javascript-yarn2nix
+
+mkYarnPackage rec {
+  pname = "draupnir";
+  version = "1.81.1";
+  # NOTE: Hard fork of Mjolnir.
+  src = fetchFromGitHub {
+    owner = "Gnuxie";
+    repo = "Draupnir";
+    rev = "v${version}";
+    sha256 = "RyVTdGnt4LbOyqYgSxbNpCYIFeYxgvlkDlFWN1Jw4DY=";
+  };
+
+  nativeBuildInputs = [
+    nodejs
+    makeWrapper
+    pkgs.yarn
+    pkgs.matrix-sdk-crypto-nodejs
+  ];
+  offlineCache = fetchYarnDeps {
+    yarnLock = src + "/yarn.lock";
+    hash = "sha256-wXlaYC2RPPTSwCkuw3TsRAMV5sKsZrJDdanj16p1CXA=";
+  };
+  yarnLock = src + "/yarn.lock";
+  packageJSON = src + "/package.json";
+
+  buildPhase = ''
+      echo "Running preBuild"
+      runHook preBuild
+      export HOME=$(mktemp -d)
+      echo "Building..."
+      yarn --offline --verbose build
+      # copy built modules to package...
+      echo "Copying built matrix-sdk-crypto-nodejs modules to package..."
+      cp -a ${pkgs.matrix-sdk-crypto-nodejs}/lib/node_modules/* node_modules/
+      echo "Adding version.txt..."
+      mkdir -p deps/draupnir/
+      echo "${version}-nix" > deps/draupnir/version.txt
+      echo "Running postBuild..."
+      runHook postBuild
+    '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/share
+    cp -a . $out/share/draupnir
+
+    #include config
```

Comment:
?

Reply:
Not sure what's confusing here?

Reply:
I see `#include` and the first place my brain goes is "is this code being run through a C preprocessor?"

Slowing down and reading the context it makes sense, but maybe adding a space after the `#` would prevent that knee-jerk reaction.

Reply:
Ah, right. Makes sense